index-providers/modeshape-elasticsearch-index-provider/src/main/java/org/modeshape/jcr/index/elasticsearch/EsIndexColumn.java

```diff
@@ -0,0 +1,254 @@
+    }
+
+    /**
+     * Gets the name of the field to store length.
+     * 
+     * @return 
+     */
+    public String getLengthFieldName() {
+        return LENGTH_PREFIX + name;
+    }
+    
+    public String getLowerCaseFieldName() {
+        return LOWERCASE_PREFIX + name;
+    }
+
+    public String getUpperCaseFieldName() {
+        return UPPERCASE_PREFIX + name;
+    }
+    
+    /**
+     * Gets type of this column.
+     * 
+     * @return Jcr type of this column.
+     */
+    public PropertyType getType() {
+        return type;
+    }
+
+    /**
+     * Gets Elasticsearch core type for this column.
+     *
+     * @return ElasticSearch core type name.
+     */
+    public String getElasticSearchType() {
+        switch (type) {
+            case BINARY:
+                return "binary";
+            case BOOLEAN:
+                return "boolean";
+            case DATE:
+                return "long";
+            case LONG:
+            case DECIMAL:
+                return "long";
+            case DOUBLE:
+                return "double";
+            default:
+                return "String";
+        }
+    }
+
+    /**
+     * Appends column definition to Elasticsearch mapping.
+     * 
+     * @param builder Mapping builder.
+     * 
+     * @throws IOException 
+     */
+    protected void append(XContentBuilder builder) throws IOException {
+        switch (type) {
+            case BINARY:
+                builder.startObject(name).field("type", "binary").endObject();
+                break;
+            case BOOLEAN:
+                builder.startObject(name).field("type", "boolean").endObject();
+                break;
+            case DATE:
+                builder.startObject(name).field("type", "long").endObject();
+                break;
+            case LONG:
+            case DECIMAL:
+                builder.startObject(name).field("type", "long").endObject();
+                break;
+            case DOUBLE:
+                builder.startObject(name).field("type", "double").endObject();
+                break;
+            default:
+                builder.startObject(name).field("type", "string")
+                        .field("analyzer", "whitespace").endObject();
+                break;
+        }
+    }
+
+    /**
+     * Converts representation of the given value using type conversation rules
+     * between JCR type of this column and Elasticsearch core type of this column.
+     * 
+     * @param value value to be converted.
+     * @return converted value.
+     */
+    protected Object columnValue(Object value) {
+        switch (type) {
+            case PATH:
+            case NAME:
+            case STRING:
+            case REFERENCE:
+            case SIMPLEREFERENCE:
+            case WEAKREFERENCE:
+            case URI:
+                return valueFactories.getStringFactory().create(value);
+            case DATE :
+                return ((DateTime) value).getMilliseconds();
```

Comment:
There is no `default` case here. What about longs or binaries or other `PropertyType`s?